cmd/flightctl-api/main.go

```diff
@@ -30,17 +32,29 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-//nolint:gocyclo
 func main() {
+	startTime := time.Now()
 	ctx := context.Background()
 
+	ctx, cancel := signal.NotifyContext(ctx, os.Interrupt, syscall.SIGHUP, syscall.SIGTERM, syscall.SIGQUIT)
+	defer cancel()
+
+	if err := runCmd(ctx, startTime, cancel); err != nil {
+		log.InitLogs().Fatalf("flightctl-api: %v", err)
+	}
+}
+
+//nolint:gocyclo
+func runCmd(ctx context.Context, startTime time.Time, cancel context.CancelFunc) error {
```

Comment:
no need to pass cancel lets just make a child ctx in run

```suggestion
func runCmd(ctx context.Context) error {
	startTime := time.Now()
    ctx, cancel := context.WithCancel(ctx)
    defer cancel()
```